src/Symfony/Component/Yaml/Parser.php

```diff
@@ -1137,106 +1141,148 @@ private function getLineTag(string $value, int $flags, bool $nextLineCheck = tru
-        }
+        return $this->lexInlineStructure($cursor, '}');
+    }
 
-        for ($i = 1; isset($yaml[$i]) && ']' !== $yaml[$i]; ++$i) {
-        }
+    private function lexInlineSequence(int &$cursor = 0): string
+    {
+        return $this->lexInlineStructure($cursor, ']');
+    }
 
-        if (isset($yaml[$i]) && ']' === $yaml[$i]) {
-            return $yaml;
-        }
+    private function lexInlineStructure(int &$cursor, string $closingTag): string
+    {
+        $value = $this->currentLine[$cursor];
+        ++$cursor;
 
-        $value = $yaml;
+        do {
+            $this->consumeWhitespaces($cursor);
+
+            while (isset($this->currentLine[$cursor])) {
+                switch ($this->currentLine[$cursor]) {
+                    case '"':
+                    case "'":
+                        $value .= $this->lexInlineQuotedString($cursor);
+                        break;
+                    case ':':
+                    case ',':
+                        $value .= $this->currentLine[$cursor];
+                        ++$cursor;
+                        break;
+                    case '{':
+                        $value .= $this->lexInlineMapping($cursor);
+                        break;
+                    case '[':
+                        $value .= $this->lexInlineSequence($cursor);
+                        break;
+                    case $closingTag:
+                        $value .= $this->currentLine[$cursor];
+                        ++$cursor;
+
+                        return $value;
+                    case '#':
+                        break 2;
+                    default:
+                        $value .= $this->lexUnquotedString($cursor);
+                }
 
-        while ($this->moveToNextLine()) {
-            for ($i = 1; isset($this->currentLine[$i]) && ']' !== $this->currentLine[$i]; ++$i) {
+                if ($this->consumeWhitespaces($cursor)) {
+                    $value .= ' ';
+                }
             }
 
-            $trimmedValue = trim($this->currentLine);
+            $cursor = 0;
+        } while ($this->moveToNextLine());
 
-            if ('' !== $trimmedValue && '#' === $trimmedValue[0]) {
-                continue;
-            }
+        return $value;
+    }
+
+    private function consumeWhitespaces(int &$cursor): bool
+    {
+        $whitespacesConsumed = 0;
 
-            $value .= $trimmedValue;
+        do {
+            $whitespaceOnlyTokenLength = strspn($this->currentLine, ' ', $cursor);
+            $whitespacesConsumed += $whitespaceOnlyTokenLength;
+            $cursor += $whitespaceOnlyTokenLength;
 
-            if (isset($this->currentLine[$i]) && ']' === $this->currentLine[$i]) {
-                break;
+            if (isset($this->currentLine[$cursor])) {
+                return 0 < $whitespacesConsumed;
             }
-        }
 
-        return $value;
+            $cursor = 0;
```

Comment:
Here a problem for wrong invalid yaml documents as well (reproduced it if you need). Might be relevant in valid cases as well.

The last one is in `lexInlineMapping`. I didn't check if this can cause harm but at least it shouldn't set the cursor back there as well imo.